MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -266,12 +267,25 @@ - (void)applyEnabledState:(BOOL)isEnabled {
 
   // Enabling
   if (isEnabled) {
+    id<MSCrashHandlerSetupDelegate> crashSetupDelegate = [MSWrapperCrashesHelper getCrashHandlerSetupDelegate];
 
-    // Check if there is a wrapper SDK that needs to do some custom handler setup. If there is,
-    // then the wrapper SDK will call [self configureCrashReporter].
-    if (![[MSWrapperExceptionManager getDelegate] respondsToSelector:@selector(setUpCrashHandlers)] ||
-        ![[MSWrapperExceptionManager getDelegate] setUpCrashHandlers]) {
-      [self configureCrashReporterWithUncaughtExceptionHandlerEnabled:YES];
+    // Check if a wrapper SDK has a preference for uncaught exception handling
+    BOOL enableUncaughtExceptionHandler = YES;
+    if ([crashSetupDelegate respondsToSelector:@selector(shouldEnableUncaughtExceptionHandler)]) {
+      enableUncaughtExceptionHandler = [crashSetupDelegate shouldEnableUncaughtExceptionHandler];
+    }
+
+    // Allow a wrapper SDK to perform custom behavior before setting up crash handlers
+    if ([crashSetupDelegate respondsToSelector:@selector(willSetUpCrashHandlers)]) {
+      [crashSetupDelegate willSetUpCrashHandlers];
+    }
+
+    // Set up crash handlers
+    [self configureCrashReporterWithUncaughtExceptionHandlerEnabled:YES];
```

Comment:
Shouldn't this take `enableUncaughtExceptionHandler`? This looks like the uncaught exception handler will be enabled no matter what the `CrashSetupDelegate` implements.